notifications.bs

```diff
@@ -601,6 +604,39 @@ or vibrating the device again, unless the <a>renotify preference flag</a> is set
 </ol>
 
 
+<h3 id=selecting-an-icon-url>Selecting an icon URL</h3>
+
+<p>The <dfn>icon URL selection steps</dfn> for selecting a URL given a list <var>icons</var> of type
+{{NotificationIcon}}, a <var>fallbackURLString</var> of type {{USVString}}, and
+<a lt="base url">base URL</a> <var>baseURL</var> are:
+
+<ol>
+  <li><p>Let <var>selectedURL</var> and <var>selectedIcon</var> be null
+
+  <li><p>For each <var>icon</var> in <var>icons</var> run these substeps:
+
+  <ol>
+    <li><p>Let <var>iconURL</var> be the result of <a lt="url parser">parsing</a> <var>icon</var>'s
+    <code>src</code> using <var>baseURL</var>. If <var>iconURL</var> is failure, abort these
+    substeps for this <var>icon</var>
+
+    <li><p>Let <var>parsedSizes</var> be the result of <a lt="attr link sizes">parsing</a>
+    <var>icon</var>'s <code>sizes</code>
+
+    <li><p>If <var>icon</var> is considered by the implementation to be more suitable than
+    <var>selectedIcon</var>, taking into account whether <var>icon</var>'s <code>type</code> is
+    <a lt="valid mime type">valid</a> and <a lt="resources">supported</a>, as well as the value of
+    <var>parsedSizes</var>, then set <var>selectedURL</var> to <var>iconURL</var> and set
+    <var>selectedIcon</var> to <var>icon</var>
```

Comment:
I wrote it, but I'm not sure whether that's what you're asking. It's similar to bits of the manifest spec, but simplified, as WebIDL does a lot of the defining for us, and we don't need to describe how to parse JSON.